speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontendUnitTests/IndicatorConfigTests.cpp

```diff
@@ -1,7 +1,351 @@
+void AssertSound(tSoundData* p_loadedSnd, tSoundData* p_rndSnd)
+{
+    // If the generated sound is null, only check whether the loaded sound is also null
+    if (!p_rndSnd)
+    {
+        ASSERT_EQ(p_loadedSnd, p_rndSnd);
+        return;
+    }
+
+    char* rndSndFullPath = new char[PATH_BUF_SIZE];
+    snprintf(rndSndFullPath, PATH_BUF_SIZE, SOUNDS_DIR_FORMAT, GfDataDir(), p_rndSnd->Path);
+    ASSERT_STREQ(p_loadedSnd->Path, rndSndFullPath);
+    ASSERT_EQ(p_loadedSnd->Looping, p_rndSnd->Looping);
+    ASSERT_EQ(p_loadedSnd->LoopInterval, p_rndSnd->LoopInterval);
+}
+
+/// @brief             Asserts whether the loaded texture is equal to the generated random texture.
+/// @param p_loadedTex The texture loaded into the indicator config
+/// @param p_rndTex    The randomly generated texture
+void AssertTexture(tTextureData* p_loadedTex, tTextureData* p_rndTex)
+{
+    // If the generated texture is null, only check whether the loaded texture is also null
+    if (!p_rndTex)
+    {
+        ASSERT_EQ(p_loadedTex, p_rndTex);
+        return;
+    }
+
+    ASSERT_STREQ(p_loadedTex->Path, p_rndTex->Path);
+    ASSERT_EQ(p_loadedTex->ScrPos.X, p_rndTex->ScrPos.X);
+    ASSERT_EQ(p_loadedTex->ScrPos.Y, p_rndTex->ScrPos.Y);
+}
+
+/// @brief             Asserts whether the loaded text is equal to the generated random text.
+/// @param p_loadedTxt The text loaded into the indicator config
+/// @param p_rndTxt    The randomly generated text
+void AssertText(tTextData* p_loadedTxt, tTextData* p_rndTxt)
+{
+    // If the generated texture is null, only check whether the loaded texture is also null
+    if (!p_rndTxt)
+    {
+        ASSERT_EQ(p_loadedTxt, p_rndTxt);
+        return;
+    }
+
+    ASSERT_STREQ(p_loadedTxt->Text, p_rndTxt->Text);
+    ASSERT_EQ(p_loadedTxt->ScrPos.X, p_rndTxt->ScrPos.X);
+    ASSERT_EQ(p_loadedTxt->ScrPos.Y, p_rndTxt->ScrPos.Y);
+}
+
+/// @brief                    Asserts whether the loaded indicator is equal to the generated indicator
+/// @param p_loadedIndicator  The indicator loaded into the indicator config
+/// @param p_rndIndicator     The randomly generated indicator
+void AssertIndicator(tIndicatorData p_loadedIndicator, tIndicatorData p_rndIndicator)
+{
+    ASSERT_EQ(p_loadedIndicator.Action, p_rndIndicator.Action);
+    AssertSound(p_loadedIndicator.Sound, p_rndIndicator.Sound);
+    AssertTexture(p_loadedIndicator.Texture, p_rndIndicator.Texture);
+    AssertText(p_loadedIndicator.Text, p_rndIndicator.Text);
+}
+
+/// @brief Creates a random indicator data object, writes this to an xml file,
+///        then reads in this xml file using IndicatorConfig::LoadIndicatorData
+///        Asserts whether the loaded in data is equal to the randomly generated data.
+TEST(IndicatorConfigTests, LoadIndicatorDataFromXML)
+{
+    // Gf module contains all the XML reading/writing functions.
+    GfInit(GF_LOGGING_DISABLE);
+    GfSetDataDir(SD_DATADIR_SRC);
+    ASSERT_TRUE(SetupSingletonsFolder());
+    Random rnd;
```

Comment:
Would it be possible to log the seed that is being used so it's easier to replicate a fail?

Reply:
I will look into this using `SetUp` and `TearDown` functionality of GTest. I think it should be possible to log the random seed used in every test.